src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -24,7 +24,10 @@ public class TagHelperRunner
                 executionContext.AllAttributes,
                 executionContext.UniqueId,
                 executionContext.GetChildContentAsync);
-            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes);
+            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes)
+            {
+                SelfClosing = executionContext.SelfClosing,
```

Comment:
@yishaigalatzer no there's not with this fix in mind. We will always need to be explicit as to if the output is self closing or not since that info will come from what the user originally typed.